src/main/java/hudson/plugins/scm/koji/KojiSCM.java

```diff
@@ -143,9 +142,10 @@ public void checkout(Run<?, ?> run, Launcher launcher, FilePath workspace, TaskL
             throw new RuntimeException("Expected instance of KojiRevisionState, got: " + baseline);
         }
 
-        // TODO add some flag to allow checkout on local or remote machine
-        KojiBuildDownloader downloadWorker = new KojiBuildDownloader(createConfig(), createNotProcessedNvrPredicate
-                (run.getParent()), run.getCause(Cause.UserIdCause.class) != null);
+        File checkoutBuildFile = new File(run.getParent().getRootDir(), BUILD_XML);
```

Comment:
This is always processed on master. It reads build.xml from job folder and pass it to the downloader, which is then executed on correct workspace